ARMmbed/mbedtls
# Add conditional dependency to tests

## Description
Some tests can run only if some compile time definitions meet a certain condition. For example, 8192 bit rsa key tests should not run if `MBEDTLS_MPI_MAX_SIZE` is less than 1024.
This PR adds a way to check compile time definitions values, for determining
whether to skip tests.

## Status
**READY**

## Requires Backporting
This is an enhancemnet, but to hte test infrastructure, so we may consider backporting this


## Additional comments
Tested on Linux and K64F

## Todos


## Steps to test or reproduce
Set `MBEDTLS_MPI_MAX_SIZE` to 512, and run `test_suite_rsa`. The test "RSA Check Public key #6 (N exactly 8192 bits)" should be skipped
